corehq/ex-submodules/couchforms/analytics.py

```diff
@@ -77,7 +77,16 @@ def app_has_been_submitted_to_in_last_30_days(domain, app_id):
     return iso_string_to_datetime(result[0]['received_on']) > (now - _30_days)
 
 
+@quickcache(['domain'], memoize_timeout=0, timeout=12 * 3600)
 def get_all_xmlns_app_id_pairs_submitted_to_in_domain(domain):
+    """This is used to get (XMLNS, app_id) from submitted forms. The results
+    get combined with data from all current app versions which means
+    that this is only used to get (XMLNS, app_id) combos from forms submitted
+    in the past which no longer have a corresponding form in the app (e.g. form deleted)
+
+    Given that we can cache this for a long period of time under the assumption that
+    a user isn't going to submit a form and then delete it from their app immediately.
+    """
```

Comment:
Haha, I have to say... _someone_ is going to do this and then be really confused. What if deleting a form / module / app (/ build?) invalidated this cache (for the whole domain)? I _think_ it's simple enough to implement (though cache invalidation always starts out _seeming_ simple...), but I'll just leave it up to you to decide.

Reply:
1333aee